code/datums/noobmode.dm

```diff
@@ -0,0 +1,129 @@
+/datum/action/equipHazard
+	name = "Equip skinsuit"
+	desc = "hazard gear and internals. Takes a few seconds."
+	icon_icon = 'icons/obj/clothing/suits.dmi'
+	button_icon_state = "skinsuit"
+	var/obj/item/clothing/suit/space/skinsuit/suitslot
+	var/obj/item/clothing/mask/breath/maskslot
+	var/obj/item/clothing/head/helmet/space/helmetslot
+	var/obj/item/tank/internals/airslot
+	var/obj/item/storage/box/survBox
+	var/paniced = FALSE
+
+/datum/action/equipHazard/Trigger()
+	if(ishuman(owner))
+		var/mob/living/carbon/human/H = owner
+		testing(paniced)
+		if(!paniced)
+			if(pannickEquip(H))
+				toggle()
+		else
+			if(stowEquipment(H))
+				toggle()
+		//var/obj/item/held = H.get_active_held_item()
+		/*if(!held)
+			held = H.get_inactive_held_item()
+			if(held && istype(held,/obj/item/storage/box/survival))
+				to_chat(H,"<span class='warning'>You are putting on the outfit inside [held], hold still!</span>")
+				pannickEquip(H)
+			else
+				to_chat(H,"<span class='warning'> you must be holding the box!</span>")
+				return
+		else
+			if(held && istype(held,/obj/item/storage/box/survival))
+				to_chat(H,"<span class='warning'>You are putting on the outfit inside [held], hold still!</span>")
+				pannickEquip(H)
+				*/
+
+
+/datum/action/equipHazard/proc/GatherItems(obj/item/storage/CNT)
+	var/stuff = CNT.contents
+	airslot = locate(/obj/item/tank/internals) in stuff
+	helmetslot = locate(/obj/item/clothing/head/helmet/space) in stuff
+	maskslot = locate(/obj/item/clothing/mask/breath) in stuff
+	suitslot = locate(/obj/item/clothing/suit/space/skinsuit) in stuff
+	if(!airslot || !helmetslot || !maskslot || !suitslot)
+		return FALSE
+	return TRUE
+
+/datum/action/equipHazard/proc/toggle()
+	if(paniced)
+		paniced = 0
+	else
+		paniced = 1
+
+/datum/action/equipHazard/Grant(mob/user, obj/item/containBox)
+	. = ..()
+	survBox = containBox
+
+/datum/action/equipHazard/proc/stowEquipment(mob/living/carbon/human/USR,speed = 10)
+	to_chat(USR,"<span class='notice'> You stuff the emergency equipment back into the box")
+	if(!do_after(USR, speed))
+		return FALSE
+	testing("time passed")
+	if(helmetslot && helmetslot == USR.head)
+		survBox.attackby(helmetslot,USR)
+	if(maskslot && maskslot == USR.wear_mask)
+		survBox.attackby(maskslot,USR)
+	if(suitslot && suitslot == USR.wear_suit)
+		if(!suitslot.rolled_up)
+			suitslot.attack_self(USR)
+		survBox.attackby(suitslot,USR)
+	if(airslot && airslot.loc != survBox)//instead of checking every pocket we just check the turf
```

Comment:
```suggestion
	if(helmetslot == USR.head)
		survBox.attackby(helmetslot,USR)
	if(maskslot == USR.wear_mask)
		survBox.attackby(maskslot,USR)
	if(suitslot == USR.wear_suit)
		if(!suitslot.rolled_up)
			suitslot.attack_self(USR)
		survBox.attackby(suitslot,USR)
	if(airslot && airslot.loc != survBox)//instead of checking every pocket we just check the turf
```

Reply:
Null checks are unneeded

Reply:
Problem is if the USR is not wearing anything its also null, and NULL == NULL